pycodb/utils.py

```diff
@@ -54,16 +56,47 @@ def records_request(method, table, view, params = None, data = None):
     return perform_request(method, url, data)
 
 
-def perform_request(method, url, data):
-    '''Performs a NocoDB request'''
-    result = requests.request(method  = method,
-                              url     = url,
-                              json    = data,
-                              headers = AUTH_HEADER,
-                              timeout = 5)
-
-    if result.status_code < 200 or result.status_code > 299:
-        raise NocoDBRequestError(status_code= result.status_code,
-                                 message=result.text)
+def perform_request(
+        method: str,
+        url: str,
+        data: Optional[Dict[str, Any]] = None,
+        max_retries: int = 3,
```

Comment:
Default value of the retries number for the function should be `1` to avoid the unexpected behavior